webapp/components/signup_user_complete.jsx

```diff
@@ -295,22 +295,6 @@ export default class SignupUserComplete extends React.Component {
         }
 
         const providedPassword = ReactDOM.findDOMNode(this.refs.password).value;
-        if (!providedPassword || providedPassword.length < Constants.MIN_PASSWORD_LENGTH) {
```

Comment:
Those settings could be added to the [client config](https://github.com/mattermost/platform/blob/master/utils/config.go#L194) then. I don't think there's any problem with the server leaking the password requirements to unauthenticated users